### ISSUE TITLE
AsyncTask and new version

### ISSUE THREAD

**Issue by arjunmenon** [NanoHttpd/nanohttpd#485]
Hey
I got two questions
1. If server is running and later the phone sleeps, the app stops responding to requests. I searched around and the solution for that is to push the service in AsyncTask.

Since not being a Android specialist, can you give a simple example for this; where the server would not stop at all even if the phone sleeps and the phone has battery saver enabled?

I have seen in my phone the native Music player never stops the whole night. It keeps running. So it is possible to achieve this with the server as well.

2.  Can you upload the latest version of Nanohttpd to Maven? I can't get it from Sonatype. It is failing.

**Comment by immosk** [comment]
no response? I like to know this case too

**Comment by LordFokas** [comment]
I don't think you can get an answer on the first one any time soon, as I don't know much about android.

As for the second, until the other members become active again you'll have to make do with what's available, unfortunately. They are both missing in action and no one gave me any access to anything besides the ability to push to this repository, so I can't do anything about it...

If things keep going like this I might just make my fork the "official"(-ish) one and just keep things running my way over there, but until I have some more free time myself that's not gonna happen either.

**Comment by arjunmenon** [comment]
Maybe you can introduce an alternate branch in maven, which tracks all the latest commits. Many of those pull requests are pending since ages and they look important and interesting. We can atleast start merging them one by one.

**Comment by ashutoshgngwr** [comment]
Using [WakeLock](https://developer.android.com/reference/android/os/PowerManager.WakeLock.html) with a [Foreground Service](https://developer.android.com/guide/components/services.html) would be a better solution. Although, with doze mode on Android Marshmallow and upwards, wake locks are ignored when the device enters doze mode. Official docs on Doze and App Standby restrictions can be found [here](https://developer.android.com/training/monitoring-device-state/doze-standby.html).

**Comment by arjunmenon** [comment]
@ashutoshgngwr It needs to be a background service. The server would respond to requests as and when arrived.

**Comment by ashutoshgngwr** [comment]
Both a background as well as foreground service on Android runs on the main thread. So any android service would run its time consuming task (server daemon in this case) on a separate thread. Foreground here only signifies that the use is notified/aware of the task running, and prevent Android from killing the service in case of low memory. More info can be found [here](https://developer.android.com/reference/android/app/Service.html#startForeground(int,%20android.app.Notification)).
